cpp/tests/interop/nanoarrow_utils.hpp

```diff
@@ -183,6 +204,24 @@ void get_nanoarrow_dict_array(ArrowArray* arr,
   get_nanoarrow_array<IND_TYPE>(arr, ind, validity);
 }
 
+template <typename KEY_TYPE, typename IND_TYPE>
+void populate_dict_from_col(ArrowArray* arr, cudf::dictionary_column_view dview)
```

Comment:
We should be consistent about whether these helpers are defined in this header or in `to_arrow_device_test.cpp`. `populate_list_from_col` is declared here but defined there. Do you think they're reusable (also possibly a question for @davidwendt on his PR)? If they are, maybe we need a `nanoarrow_utils.cpp` file with definitions, which I also think David has created in his PR.

Reply:
I'm not sure how much effort we should put into supporting dictionary.
I would be ok with throwing an exception for now and then supporting it later when someone asks for it.
This goes for `to_arrow_device` as well.